java/server/src/org/openqa/grid/web/servlet/api/v1/utils/ProxyUtil.java

```diff
@@ -0,0 +1,121 @@
+// Licensed to the Software Freedom Conservancy (SFC) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The SFC licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.openqa.grid.web.servlet.api.v1.utils;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import org.openqa.grid.internal.RemoteProxy;
+import org.openqa.grid.internal.TestSession;
+import org.openqa.grid.internal.TestSlot;
+import org.openqa.selenium.remote.CapabilityType;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ProxyUtil {
+
+  private static final String TOTAL = "total";
+  private static final String BUSY = "busy";
+
+  private ProxyUtil() {
+    //defeat instantiation.
+  }
+
+  /**
+   * @param proxy - A {@link RemoteProxy} instance whose details are being sought.
+   * @return - A {@link JsonObject} that contains the details of the remote proxy.
+   */
+  public static JsonObject getNodeInfo(RemoteProxy proxy) {
+    JsonObject computer = new JsonObject();
+    computer.addProperty("host", proxy.getRemoteHost().getHost());
+    computer.addProperty("port", proxy.getRemoteHost().getPort());
+    computer.addProperty("id", proxy.getId());
```

Comment:
I liked it better when we were hashing the `Id` for the benefit of the HTTP URL. This looks silly to me `http://localhost:4444/api/v1/proxy/http://192.168.1.5:5555` due to to `http://` ... `http://`